rpc/src/responses/block_ids_in_chain.rs

```diff
@@ -19,6 +20,66 @@ impl Response for BlocksInChainResponse {
 #[cfg(test)]
 mod test {
     use super::*;
+    use serde::{Deserialize, Serialize};
+
+    /// Mock struct that represents the
+    /// object expected to be found in
+    /// a response.
+    #[derive(Serialize, Deserialize)]
+    struct MockBlockInChain {
+        block_id: String,
+    }
+
+    struct BlocksInChainMockResponse {
+        pub mock_block_ids: json_array::JsonArray<MockBlockInChain>,
+    }
+
+    impl Response for BlocksInChainMockResponse {
+        /// Parses a response string in the form
+        /// `"[[{"block_id":"alpha_numeric_block_id_string"}], ["..."]]"` or
+        /// `"[{"block_id":"alpha_numeric_block_id_string"}, ...]"` into a
+        /// [`BlocksInChainMockResponse`](Self).
+        fn from_response_str(response: &str) -> Result<Self, ParseError> {
+            let mock_block_ids = json_array::JsonArray::from_str(response)?;
+            Ok(Self { mock_block_ids })
+        }
+    }
+
+    #[test]
+    fn get_blocks_in_chain_from_object_response_expecting_string_fails() {
+        let mock_response = r#"[[{"block_id": "1"}]]"#;
+
+        let blocks_response = BlocksInChainResponse::from_response_str(mock_response);
+        assert!(blocks_response.is_err());
+    }
+
+    #[test]
+    fn get_blocks_in_chain_from_object_response_expecting_object_ok() {
+        let mock_response = r#"[[{"block_id": "1"}]]"#;
+
+        let blocks_response = BlocksInChainMockResponse::from_response_str(mock_response);
+        assert!(blocks_response.is_ok());
+
+        let blocks = blocks_response.unwrap().mock_block_ids.into_vec();
+        assert!(blocks.len() == 1);
+
+        let single_block = &blocks[0];
+        assert!(single_block.block_id == "1");
```

Comment:
The indexed vec notation makes me very nervous in general (rust specific), and looks kind of funky with the borrow anyway. Maybe a `pop` or similar could be more clear.

Reply:
Good call on using `pop` but this test will be removed (explanation in other comments)